docs/BackendAPI.md

```diff
@@ -0,0 +1,52 @@
+# ONNX Backend Interface Proposal
+
+We propose a cross-platform API for loading and executing ONNX graphs on optimized backends. High-level frameworks and applications can use this API to execute neural network and machine learning models. Hardware vendors can implement this API to expose specialized hardware accelerators and highly optimized software infrastructure to the users.
+
+## Core Features
+
+- Standardized interface for neural network inference on special-purpose accelerators, CPUs, GPUs, DSPs, and FPGAs
+- Based on widely supported technologies
+  - C API for function calls
+  - ONNX format for passing model graphs
+  - NCHW tensor layout for passing inputs and outputs
```

Comment:
Why limit to NCHW? There's frameworks that have limited support for that. Why not allow users to specify the tensor layout, and then the underlying implementation can update layout as needed?

Reply:
Mostly for simplicity and since current ONNX ops assume NCHW. From our experience - we usually offload the entire subgraph to backend and it can change the layout of tensors in the bigger chunk internally.

Reply:
Note that de-facto it affects only input and output tensors, because within the graph the backend library is free to use any representation. Frameworks which favor different tensor layouts can still use the Backend API by inserting ONNX **Transpose** op before all inputs and after all outputs.

Reply:
Note that running transpose has a perf toll that would be great to avoid if possible.
However, this is not a one-way-door, and we can always revisit and add support for NHWC in the future.